Lib/test/test_type_comments.py

```diff
@@ -0,0 +1,203 @@
+import ast
+import unittest
+
+
+funcdef = """\
+def foo():
+    # type: () -> int
+    pass
+
+def bar():  # type: () -> None
+    pass
+"""
+
+asyncdef = """\
+async def foo():
+    # type: () -> int
+    return await bar()
+
+async def bar():  # type: () -> int
+    return await bar()
+"""
+
+forstmt = """\
+for a in []:  # type: int
+    pass
+"""
+
+
+withstmt = """\
+with context():  # type: int
```

Comment:
This one surprised me.  As there was no name bound, I expected the comment to be ignored.

Reply:
Again, not strong opinion, but I would make this a `SyntaxError`

Reply:
I have to think about that for a bit. It would have to be a "pass 2" syntax error, because doing this by modifying the grammar (while possible) would contort the grammar for `with_stmt`. And ruling this out would really be up to the type checker or linter that's interpreting type comments -- the syntactic rules for type comments are intentionally weak. (Just like for annotations, `a: str = 0` should not be a runtime error.)

FWIW, I am wondering if it was a mistake to even allow type comments on for- and with-statements, since these have no equivalent type annotation. (There's even a note in PEP 526 about this.)

Reply:
Yes, leaving this up to a linter/type-checker sounds OK as well.